plugins/search-backend/src/types.ts

```diff
@@ -14,6 +14,8 @@
  * limitations under the License.
  */
 
+import { IndexableDocument } from '@backstage/plugin-search-indexer-backend';
```

Comment:
Intresting reference:
https://docs.oracle.com/cd/E29597_01/fa_apirefs.1111/e21908/oracle/ecsf/IndexableDocument.html

Oracle Fusion has a similar object name for a related purpose,

> public interface IndexableDocument extends Document
>
>An indexable document represents an instance of a searchable object that can be indexed by a search engine.
>
> search engine implementation should have its own implementation of this interface to represent its internal data structure and expose it per this interface.

Just sharing, no feedback per se